src/rust/engine/workunit_store/src/lib.rs

```diff
@@ -508,13 +525,19 @@ pub struct HeavyHittersInnerStore {
 fn first_matched_parent(
   workunit_records: &HashMap<SpanId, Workunit>,
   mut span_id: Option<SpanId>,
+  is_terminal: impl Fn(&Workunit) -> bool,
```

Comment:
What does terminal mean here? Imo it's ambiguous if it's something like "final/complete" vs. "the CLI/terminal"